cmd/bucket-encryption-handlers.go

```diff
@@ -84,6 +87,19 @@ func (api objectAPIHandlers) PutBucketEncryptionHandler(w http.ResponseWriter, r
 		writeErrorResponse(ctx, w, errorCodes.ToAPIErr(ErrKMSNotConfigured), r.URL)
 		return
 	}
+	kmsKey := encConfig.KeyID()
+	if kmsKey != "" {
+		kmsContext := kms.Context{"MinIO admin API": "ServerInfoHandler"} // Context for a test key operation
+		_, err := GlobalKMS.GenerateKey(kmsKey, kmsContext)
+		if err != nil && !errors.Is(err, kes.ErrKeyExists) {
+			if errors.Is(err, kes.ErrKeyNotFound) {
+				writeErrorResponse(ctx, w, toAPIError(ctx, kms.KeyNotFoundException{KeyID: kmsKey}), r.URL)
+				return
+			}
+			writeErrorResponse(ctx, w, toAPIError(ctx, err), r.URL)
+			return
+		}
+	}
```

Comment:
this part could be written as a function and then re-used in serverInfohandler as well @poornas

Reply:
the intent here is to just prove existence of the kms key, where as the admin info handler is interested in proving kms reachable and capable of (enc/dec)rypting - didn't find a reusable function that fits in naturally.

Reply:
Yeah, but why not do that here as well @poornas we don't need to use a "key" that can't do enc/decrypt - it is natural to have a function that tells us "yes/no" here. 

It is an additional benefit to know if "key" exists or not. 

I think this relaxed way of handling was on purpose by @aead - so that users can `mc admin kms create` the key while we allow them to set a non-existent key during PutBucketEncryption() call.

Reply:
All errors internal to KES cannot be converted to API errors, the idea was to be able to give a meaningful error when kms key id is invalid for put-object requests.

Reply:
Okay then its fine.